apps/SageAccountWeb/src/hooks/useMaybeRedirectToSignTermsOfService.ts

```diff
@@ -23,7 +23,7 @@ export default function useMaybeRedirectToSignTermsOfService(): UseMaybeRedirect
   const location = useLocation()
 
   // true until we confirm we will not redirect
-  const mayRedirect = useRef(true)
+  const [mayRedirect, setMayRedirect] = useState(true)
```

Comment:
by using `useRef`, we avoided triggering re-renders when `mayRedirect` was updated. We DO want the re-render, so the redirect behavior in the LoginRedirector component is retriggered when this finally resolves to `false`.